GetBestStudents/best_students/StudentListCreater.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Text.Json;
+
+namespace BEST_STUDENTS;
+
+public class StudentListCreater
+{
+    public List<Student> CreateStudentsList(JsonElement studentRoot, int studentCount)
+    {
+        List<Student> students = new List<Student>();
+        for(int i = 0; i < studentCount; i++)
+        {
+            students.Add(new Student(studentRoot[i]));
+        }
+
+        return students;
+    }
+
+    public List<Student> AddStudentAverages(List<Student> students, List<JsonElement> scores)
+    {
+        foreach (var student in students)
+        {
+            List<JsonElement> tempScores = new List<JsonElement>(scores
+                .Where(x => student.ScoreIsForStudent(x)));
+
+            student.Avg = tempScores.ConvertAll(x => x.GetProperty("Score").GetDouble()).Average();
```

Comment:
I see _x_ is a member of _scores_ which is a list of JsonElements representing objects of type _Score_. (Though you do not have this class and I don't understand **why?**). 
Now each _x_ has a field _Score_ and this is somehow confusing. Wouldn't it be better to have a _Grade_ class and then define the field _Score_ for that?

Reply:
Class Grade was created and problem was solved